apm-agent-plugins/apm-mongoclient-plugin/src/test/java/co/elastic/apm/agent/mongoclient/AbstractMongoClientInstrumentationTest.java

```diff
@@ -0,0 +1,187 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.mongoclient;
+
+import co.elastic.apm.agent.AbstractInstrumentationTest;
+import co.elastic.apm.agent.impl.transaction.TraceContext;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+import org.bson.Document;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+public abstract class AbstractMongoClientInstrumentationTest extends AbstractInstrumentationTest {
+
+    @SuppressWarnings("NullableProblems")
+    protected static GenericContainer container;
+    protected static final String DB_NAME = "testdb";
+    protected static final String COLLECTION_NAME = "testcollection";
+
+    @BeforeClass
+    public static void startContainer() {
+        container = new GenericContainer("mongo:3.4").withExposedPorts(27017);
+        container.start();
+    }
+
+    @AfterClass
+    public static void stopContainer() {
+        container.stop();
+        container = null;
+    }
+
+    @Before
+    public void startTransaction() throws Exception {
+        Transaction transaction = tracer.startTransaction(TraceContext.asRoot(), null, null).activate();
+        transaction.withName("Mongo Transaction");
+        transaction.withType("request");
+        transaction.withResultIfUnset("success");
+    }
+
+    @After
+    public void endTransaction() throws Exception {
+        try {
+            Transaction currentTransaction = tracer.currentTransaction();
+            if (currentTransaction != null) {
+                currentTransaction.deactivate().end();
+            }
+        } finally {
+            reporter.reset();
+        }
+        dropCollection();
```

Comment:
This should be tested as well. Maybe move the collection creation to the `@BeforeClass` and do within a transaction, and drop the collection within a transaction in `@AfterClass`?